### DISCUSSION TITLE
Does bytewax replace celery?

### DISCUSSION THREAD

**Discussion by SnoozeFreddo** [bytewax/bytewax#511] [Q&A] [answered]
If I use Kafka, would bytewax be able to act like a task queue and thus replace celery+RabbitMQ?

**Comment by davidselassie** [comment, marked as answer]
Yes, you can use Bytewax and Kafka as a task queue. The Bytewax recovery system can keep track of the offset in the task Kafka topic to ensure that every task is processed at least once. You will need to architect wherever you store your task results so that it is idempotent, because it's possible that over failure and restart, some tasks results could be written out twice, as items are read off the Kafka topic in batches.
